cmd/containerd-shim-runhcs-v1/rootfs.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/layers"
+	"github.com/containerd/containerd/api/types"
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/mount"
+)
+
+// validateRootfsAndLayers checks to ensure we have appropriate information
+// for setting up the container's root filesystem. It ensures the following:
+// - One and only one of Rootfs or LayerFolders can be provided.
+// - If LayerFolders are provided, there are at least two entries.
+// - If Rootfs is provided, there is a single entry and it does not have a Target set.
+func validateRootfsAndLayers(rootfs []*types.Mount, layerFolders []string) error {
+	if len(rootfs) > 0 && len(layerFolders) > 0 {
+		return fmt.Errorf("cannot pass both a rootfs mount and Windows.LayerFolders: %w", errdefs.ErrFailedPrecondition)
+	}
+	if len(rootfs) == 0 && len(layerFolders) == 0 {
+		return fmt.Errorf("must pass either a rootfs mount or Windows.LayerFolders: %w", errdefs.ErrFailedPrecondition)
+	}
+	if len(layerFolders) > 0 && len(layerFolders) < 2 {
+		return fmt.Errorf("must pass at least two Windows.LayerFolders: %w", errdefs.ErrFailedPrecondition)
+	}
+	if len(rootfs) > 1 || rootfs[0].Target != "" {
```

Comment:
`len(rootfs) > 0 && (len(rootfs) > 1 || rootfs[0].Target != "")`?

Maybe it's easier to reason with something like this:
```
if len(rootfs) > 0 {
    if len(layerFolders) > 0 {
        ...
    }
    if len(rootfs) > 1 || rootfs[0].Target != "" {
        ...
    }
    return nil
}
if len(layerFolders) == 0 {
    ...
}
if len(layerFolders) < 2 {
    ...
}
return nil
```

Reply:
I've cleaned this up a bit to first validate that we only have one of rootfs/layerFolders, and then do separate checks based on which we have. I hope it's clearer now (and should fix this bug).